### ISSUE TITLE
Null sideloaded relationships including links

### ISSUE THREAD

**Issue by jaredcnance** [json-api-dotnet/JsonApiDotNetCore#248]
if `User.Profile == null` we will still return the profile link
```
/users?include=profile

{
  "relationships": {
    "profile": {
      "links": {
        "self": "/users/1/profile"
      }
    }
  }
}
```

**Comment by jaredcnance** [comment]
After discussing with dgeb, more research needs to be done to ensure this is not a client issue.

> this is a problem with Ember Data imo
> I’m strongly against implicit fetches ever being triggered
> it’s fine (and proper) to return the `self` link regardless of whether `data` is also returned

**Comment by FabianHoffmann** [comment]
Wouldn't it make sense to add `"data": null` to indicate, that there is no `profile`? That's also how I read the spec: https://jsonapi.org/format/#document-resource-object-linkage 
Just adding a `self link` gives no information about if there is a `profile` or not.

Is there an easy way to add `"data": null`?

**Comment by RKennedy9064** [comment]
@jaredcnance I'm also experiencing this issue, so I figured I would add to the open issue. I'm also using Ember Data and have relationships that can sometimes be null. When one of my relationships is null, the returned data looks like this:

```
/staff-roster?include=region

"relationships": {
    "region": {  
        "links": {  
            "related": "/api/v1/staff-roster/21/region"  
            "self": "/api/v1/staff-roster/21/relationships/region"  
        }  
    }  
}  
```

Since I'm displaying information in a paginated table, whenever the relationship is null and only the links are included, Ember will make a get request for every row which just returns:

```
{
    "data": null
}
```

I'm hoping to avoid having the front end do a get request for every relationship that's null and returns a link with no data. I'm not sure if it should be handled here in the back end, or if it should be handled with Ember Data, so I figured I would share my information regarding this issue as well. Is there an easy way to add `"data": null` in the meantime?
